app/Console/Commands/ImportAshesCampaigns.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Rogue\Console\Commands;
+
+use Carbon\Carbon;
+use League\Csv\Reader;
+use Rogue\Models\Campaign;
+use Rogue\Services\Registrar;
+use Illuminate\Console\Command;
+
+class ImportAshesCampaigns extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:ashescampaignsimport {path}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Imports legacy campaigns from Ashes based on the data in the given (link to a) CSV file.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct(Registrar $registrar)
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Make a local copy of the CSV
+        $path = $this->argument('path');
+        info('rogue:legacycampaignimport: Loading in csv from ' . $path);
+
+        $temp = tempnam(sys_get_temp_dir(), 'command_csv');
+        file_put_contents($temp, fopen($this->argument('path'), 'r'));
+
+        // Load the legacy campaigns from the CSV
+        $legacy_campaigns_csv = Reader::createFromPath($temp, 'r');
+        $legacy_campaigns_csv->setHeaderOffset(0);
+        $legacy_campaigns = iterator_to_array($legacy_campaigns_csv->getRecords());
+
+        // Import each legacy campaign
+        info('rogue:legacycampaignimport: Importing legacy campaigns...');
+
+        foreach ($legacy_campaigns as $iterator => $legacy_campaign) {
+            // Normalize all "NULL" values to null
+            foreach ($legacy_campaign as $key => $value) {
+                if ($value === "NULL") {
+                    $legacy_campaign[$key] = null;
+                }
+            }
+
+            // See if the campaign exists
+            $existing_campaign = Campaign::where('id', $legacy_campaign['campaign_id'])->orWhere('id', $legacy_campaign['run_id'])->first();
+
+            // Create the campaign if there isn't one already
+            if (! $existing_campaign) {
+                // If there is no campaign_id, set the campaign_run_id to the id.
+                if (is_null($legacy_campaign['campaign_id'])) {
```

Comment:
Does this mean that in the csv runs will not have the `campaign_id` column populated?

Reply:
@katiecrane great q! It will have the `campaign_id` column! However, after running the query, some campaigns didn't have this populated so wanted to account for these edge cases. Happy to share the CSV I'm working with too if that is helpful!

Reply:
OH or does it mean that only CURRENT runs will have the `campaign_id populated>

Reply:
@chloealee hmmmm does that mean that there are runs not attached to any campaign? I think seeing the CSV would help!

Reply:
@chloealee ah okay I see, the first 7 rows are the ones with `null`. Are these actual campaigns? Seems like one of them is the DS app and 5 of them are in Spanish?

Reply:
Yeah, these look like placeholder campaigns. I think it's safe to add them with the script, and we can delete later.